azure_storage_client.go

```diff
@@ -105,17 +105,17 @@ func (util *snowflakeAzureClient) getFileHeader(meta *fileMetadata, filename str
 	}
 
 	meta.resStatus = uploaded
-	metadata := resp.Metadata
+	metadata := withLowerKeys(resp.Metadata)
 	var encData encryptionData
 
-	_, ok = metadata["Encryptiondata"]
+	_, ok = metadata["encryptiondata"]
```

Comment:
On the other hand - this behavior is independent of us, it can change in the future or even vary among different Go versions or OSes. Headers should be treated as case insensitive, so it shouldn't matter.

Reply:
Right, so there's a risk we may not catch the moment when it changes on Go/Azure proxy handling end and such annotation would be misleading so I'm ok with keeping it as is